modules/deployment/src/main/java/org/mule/runtime/module/deployment/internal/DefaultArchiveDeployer.java

```diff
@@ -303,7 +307,13 @@ public void deployArtifact(T artifact) throws DeploymentException {
   private void addZombieApp(Artifact artifact) {
```

Comment:
Change parameter type from Artifact to DeployableArtifact and remove redundant casts

Reply:
Casts are still necessary because they are for the ArtifactDescriptor not the Artifact. Should i still change the method argument?